server/src/collection.rs

```diff
@@ -75,11 +75,60 @@ impl Collection {
     #[zbus(out_args("item", "prompt"))]
     pub async fn create_item(
         &self,
-        _properties: Properties,
-        _secret: SecretInner,
-        _replace: bool,
-    ) -> Result<(OwnedObjectPath, ObjectPath), ServiceError> {
-        todo!()
+        properties: Properties,
+        secret: SecretInner,
+        replace: bool,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
+        #[zbus(signal_emitter)] signal_emitter: zbus::object_server::SignalEmitter<'_>,
+    ) -> Result<(OwnedObjectPath, OwnedObjectPath), ServiceError> {
+        let SecretInner(session, iv, secret, _content_type) = secret;
+        let label = properties.label();
+        let attributes = properties.attributes().unwrap();
+        let manager = self.manager.lock().await;
+
+        let Some(session) = manager.session(&session) else {
+            tracing::error!("The session `{}` does not exist.", session);
+            return Err(ServiceError::NoSession(format!(
+                "The session `{}` does not exist.",
+                session
+            )));
+        };
+
+        let secret = match session.aes_key() {
+            Some(key) => oo7::crypto::decrypt(secret, &key, &iv),
+            None => zeroize::Zeroizing::new(secret),
+        };
+
+        let item = match self
+            .keyring
+            .create_item(label, &attributes, secret, replace)
+            .await
+        {
+            Ok(item) => item,
+            Err(_) => todo!(), // TODO: may be new ServiceError
+        };
```

Comment:
You can't add new ServiceError, that would not be compatible with the current specification. Instead, I would tracing::error the error message or so.

Reply:
Understood, thanks!

Reply:
Ah, I still need to return an error from here. Can't use `.map_err` either, missing a trait impl.

Reply:
You can, by using ServiceError::Zbus and then creating a custom zbus error message

Reply:
basically zbus::Error::FDO(zbus::fdo::Error::Failed(format!("Failed to do {e}")))`

Reply:
got it, thank you!